### PR TITLE
docs: permissioned contract deployment with tooling

### LINKED ISSUE #917: update permissioned deployment to match dapp-agoric-basics

### LINKED ISSUE #931: Docs Revamp - Page 4
Page 4: 
Goal: Developer can deploy their own bundles to devnet and Emerynet

Should know:

- When to use which network
- What a proposal is
- How a proposal gets into production
- The workflow, what you need for what

Testing 
in ALL environments

Debugging
in ALL environments

### LINKED ISSUE #961: Docs Revamp - Permissioned Deployments - writeup
**User Story:** As a Developer I should be able to read the docs and understand how and where I can make permissioned deployments in the Agoric ecosystem. 

- Define a checklist that I can walk thru for Devnet / EmeryNet / Mainnet
- Define the process 
- Define all the commands / steps

### PR DESCRIPTION
**Pull request by dckc**
closes #917 
closes #961 
closes #931

rendered:

 - https://dc-deploy-contract.documentation-7tp.pages.dev/guides/coreeval/ - 4 pages
 - (note a few pages outside that section are also changed: getting started, agd cli)

TODO:

 - [x] detail on devnet deployment:
   - [x] make a community post about what you're up to before deploying to devnet. Be sure to choose a novel and appropriate name for your contract during this MN-2 phase.
   - [x] you're not expected to deploy example dapps as-is to devnet. They would collide in `agoricNames.instance` if everybody did that.

builds on some recent work in dapp-agoric-basics

 - https://github.com/Agoric/dapp-agoric-basics/pull/16
 - https://github.com/Agoric/dapp-agoric-basics/pull/21

### CODE REVIEW

File: main/guides/coreeval/local-testnet.md [hunk 1]
```diff
@@ -1,45 +1,187 @@
...
-	```
+   - `import { E } from '@endo/far'`
+     A bundle strips this declaration during bundling
+     since the core-eval scope includes exports of `@endo/far`.
+   - `bundleID = ...` is replaced using updated/cached bundle hash
+   - `main` is appended as the script completion value
+   - the `permit` export is emitted to a `.json` file
 
-    For example, to deploy the PSM proposal referenced in the previous topics, run the following:
+See [rollup.config.mjs](https://github.com/Agoric/dapp-agoric-basics/blob/dc-vote-fee/contract/rollup.config.mjs) for details on using the
```
Comment (dckc) [inline main/guides/coreeval/local-testnet.md:32]: this is a link to the `dc-vote-fee` branch, which is WIP

File: main/guides/coreeval/local-testnet.md [hunk 2]
```diff
@@ -1,45 +1,189 @@
...
+
+```console
+dapp-agoric-basics/contract$ yarn bundle-source --cache-json bundles/ src/sell-concert-tickets.contract.js sell-concert-tickets
+bundles/ add: sell-concert-tickets from src/sell-concert-tickets.contract.js
+bundles/ bundled 141 files in bundle-sell-concert-tickets.json at 2024-03-09T20:48:43.784Z
+```
+
+<img src='../assets/cosgov-install-bundle.png' alt='Install Bundle screenshot' width="600" />
+
+Then drop your permit and core eval script on the **CoreEval Proposal** tab:
```
Comment (LuqiPan) [inline main/guides/coreeval/local-testnet.md:79]: _nitpick_

You probably meant to add another newline before the  tab below?

File: main/guides/coreeval/local-testnet.md (hunk 1 above)
Comment (dckc) [reply main/guides/coreeval/local-testnet.md:32]: I plan to land `dc-vote-fee` first and change the link here before landing this one.

File: main/guides/coreeval/permissions.md [hunk 3]
```diff
@@ -1,96 +1,55 @@
...
-  },
-  "instance": {
-    "consume": {
-      "economicCommittee": true
-    }
+Most contract deployments don't need everything in `BootstrapPowers`.
+Verifying by inspection that they don't use any more than they need
+is notoriously difficult. So proposals come with
+a `BootstrapManifestPermit` to declare an upper limit on the capabilities they access. For a property access `powers.P`, if the permit has `{ P: true }`, then the access succeeds. In fact, any _truthy_ value will do.
+And recursively, a property access `powers.P.Q.R` is succeeds if
```
Comment (Chris-Hibbert) [inline main/guides/coreeval/permissions.md:7]:
```suggestion
And recursively, a property access `powers.P.Q.R` succeeds if
```

File: main/guides/coreeval/permissions.md [hunk 4]
```diff
@@ -1,96 +1,55 @@
...
+  **Warning: this includes the right to over-write existing bindings to instances, brands, etc.**
 
-## Brand Section
+- **bankManager**: to manage reflection of cosmos
+  assets as ERTP assets: to register an issuer to correspond to a denom or to get a bank of purses for any address. **Warning: this includes the right to spend assets for any account.**
 
-Specifies what **[Brands](/reference/ertp-api/brand.md)**, if any, the contract produces or consumes.
\ No newline at end of file
+- **zoe**: the Zoe service.
+  **Note: this includes the right to install any available bundle and start instances of any available installation.**
```
Comment (Chris-Hibbert) [inline main/guides/coreeval/permissions.md:55]: starting instances is widely accessible.

Comment (dckc) [reply main/guides/coreeval/permissions.md:55]: only if you have access to `zoe`, right?

I suppose every contract can do `zcf.getZoeService()`. hm.

**Comment by dckc** [comment]
@sufyaankhan note changes to the list of documented [bootstrap powers](https://dc-deploy-contract.documentation-7tp.pages.dev/guides/coreeval/permissions.html#selected-bootstrappowers); for example, `feeMintAccess` is no longer discussed.

The example is now from dapp-agoric-basics. It used to be an example of adding an Inter Protocol PSM for a new token. But we now have cosgov.org tooling for that sort of thing. So it no longer seems worthwhile to document Inter Protocol powers: `economicCommitteeCreatorFacet`, `econCharterKit`, `provisionPoolStartResult`, `psmKit`. Inter Protocol is like KREAd or Crabble in this way: we don't document the bootstrap powers involved in deploying those here; we leave that to https://community.agoric.com/c/governance/core-eval/31

**Review by kbennett2000 — APPROVED** [review]

**Pull request merged**